source/codegen/templates/service_helpers.mako

```diff
@@ -352,13 +352,16 @@ ${initialize_standard_input_param(function_name, parameter)}
 
   raw_request_snippet = f'request->{field_name}_raw()'
   validate_attribute_enum = parameter.get("raw_attribute", False)
+  variable_type = "std::string" if parameter['type'] == 'char[]' else parameter['type']
 %>\
-      ${parameter['type']} ${parameter_name};
+      ${variable_type} ${parameter_name};
       switch (request->${field_name}_enum_case()) {
 % if has_unmapped_enum:
         case ${one_of_case_prefix}::k${pascal_field_name}: {
 %   if parameter['type'] in ["ViConstString", "ViString"]:
           ${parameter_name} = const_cast<${parameter['type']}>((${enum_request_snippet}).c_str());
+%   elif parameter['type'] == "char[]":
```

Comment:
Why aren't these all merged into the above cases for ViString/ViConstString? And could those all be:

`if parameter['grpc_type'] == 'string'`

?